### PR TITLE
Update variable schema with attrs field

### PR DESCRIPTION
**Pull request by CagtayFabry**
## Changes
- update schema to include `attrs` 
- `attrs` field no longer gets serialized when empty

## Related Issues
see #384

### CODE REVIEW

File: weldx/asdf/schemas/weldx.bam.de/weldx/core/variable-1.0.0.yaml [hunk 1]
```diff
@@ -34,6 +34,10 @@ properties:
     description: |
       Optional field describing the unit of the data as valid string representation.
     type: string
+  attrs:
+    description: |
+      Additional attributes metadata associated with the variable.
```
Comment (vhirtham) [inline weldx/asdf/schemas/weldx.bam.de/weldx/core/variable-1.0.0.yaml:39]:
```suggestion
      Additional attributes/metadata associated with the variable.
```

**Review by vhirtham — APPROVED** [review]

**Comment by CagtayFabry** [comment]
just noticed that the `attrs` field gets serialized when empty as `attrs: {}`

Will add a fix to correctly set `None` (and auto-remove) in that case later

**Comment by CagtayFabry** [comment]
similar to how we auto-remove `None` we could also handle other 'empty' fields such as empty dicts, might be worth considering for the future

**Comment by marscher** [comment]
Just out of curiosity, whats is the benefit of removing "empty" properties, except for saving a few bytes of metadata?

**Comment by CagtayFabry** [comment]
I think it might be more distinct/intuitive in combination with the quality standards.
With this implementation, simply adding any field to the `required` list of a schema will also ensure that there is actual data stored in the field. But I'm not 100% sure if this would not also get triggered by empty/None fields (this was also the default behavior for asdf up until 2.6 or so)

**Pull request merged**